example/src/Screens/LinkComponent.tsx

```diff
@@ -0,0 +1,140 @@
+import * as React from 'react';
+import { View, StyleSheet, ScrollView } from 'react-native';
+import { Button } from 'react-native-paper';
+import {
+  Link,
+  RouteProp,
+  ParamListBase,
+  useLinkTo,
+} from '@react-navigation/native';
+import {
+  createStackNavigator,
+  StackNavigationProp,
+} from '@react-navigation/stack';
+import Article from '../Shared/Article';
+import Albums from '../Shared/Albums';
+
+type SimpleStackParams = {
+  Article: { author: string };
+  Album: undefined;
+};
+
+type SimpleStackNavigation = StackNavigationProp<SimpleStackParams>;
+
+const LinkButton = ({
+  to,
+  ...rest
+}: React.ComponentProps<typeof Button> & { to: string }) => {
+  const linkTo = useLinkTo();
```

Comment:
Maybe it would make sense to make this return an array like `const [linkTo, props] = useLinkTo(to)` so people can utilize the complex `onPress` method created if they want. Also returning `props` could be an elegant way to strip `href` on non-web targets.

Reply:
I think maybe a separate hook will be good for it instead of overloading this one. I'm also thinking maybe `Link` can accept an `as` prop instead (ala styled components) which will serve the same purpose.

Lemme think a bit.